### DISCUSSION TITLE
Is VideoEffectDefinition supported for the Windows platform?

### DISCUSSION THREAD

**Discussion by mikesylla** [unoplatform/uno#18415] [Q&A] [answered]
Dear UNO Team,

I’m working on implementing a video editor using the MediaComposition and MediaClip classes. A key feature of the editor involves adding VideoEffectDefinitions to MediaClip for video processing. Video effect definitions are possible through the Microsoft.Graphics.Win2D NuGet package.

However, when I attempt to initialize a video effect definition, I get the exception message: "Failed to activate video effect". Everything in the implementation, including the VideoEffectDefinition interface, appears to be correct.

Could this issue be related to a lack of support for VideoEffectDefinition in UNO? I am targeting the Windows platform for this project at least for now.

Any insights or feedback would be greatly appreciated!

Michael.

**Comment by MartinZikmund** [comment, marked as answer]
You are correct, Win2D APIs are not yet generally available for Uno Platform targets. Feel free to create an issue to show s demand for the APIs you need

**Reply by mikesylla** [reply]
Thanks for the update!
